### PR TITLE
Updated plugin parent POM, README, and `Jenkinsfile`

### PR DESCRIPTION
**Pull request by janasrikanth**
I updated pom.xml, readme file and added jenkinsfile to support CD pipeline.
### Testing done

```[tasklist]
### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -5,15 +5,16 @@
   <parent>
     <groupId>org.jenkins-ci.plugins</groupId>
     <artifactId>plugin</artifactId>
-    <version>4.40</version>
+    <version>4.80</version>
     <relativePath />
   </parent>
   <artifactId>no-agent-job-purge</artifactId>
-  <version>1.3-SNAPSHOT</version>
+  <version>${changelist}</version>
```
Comment (basil) [inline pom.xml:12]: Adopting incrementals requires adding a `.mvn` directory to enable `git-changelist-maven-extension`.

Comment (janasrikanth) [reply pom.xml:12]: Thanks for the help @basil 
Added .mvn

Comment (basil) [reply pom.xml:12]: Thanks! Looks good to me.

**Review by pbuckley4192 — APPROVED** [review]

**Comment by janasrikanth** [comment]
@pbuckley4192 / @basil  can you please merge this PR

**Review by gounthar — APPROVED** [review]
Could we merge this, and maybe create a release?
Thanks.

**Pull request merged**